producer/src/main/kotlin/streams/Extensions.kt

```diff
@@ -41,4 +42,35 @@ fun ConstraintDefinition.isNodeConstraint(): Boolean {
 
 fun ConstraintDefinition.isRelationshipConstraint(): Boolean {
     return try { this.relationshipType; true } catch (e: IllegalStateException) { false }
-}
\ No newline at end of file
+}
+
+fun StreamsTransactionEvent.asSourceRecordValue(strategy: String): StreamsTransactionEvent? =
+        if(isStrategyCompact(strategy) && meta.operation == OperationType.deleted) null else this
+
+fun StreamsTransactionEvent.asSourceRecordKey(strategy: String): Any =
+        when {
+            strategy == TopicConfig.CLEANUP_POLICY_DELETE -> "${meta.txId + meta.txEventId}-${meta.txEventId}"
+            isStrategyCompact(strategy) && payload is NodePayload -> getKeyOfNode(payload as NodePayload, schema)
+            isStrategyCompact(strategy) && payload is RelationshipPayload -> getKeyOfRel(payload as RelationshipPayload)
+            else -> throw RuntimeException("Invalid kafka.streams.log.compaction.strategy value: $strategy")
```

Comment:
This is a runtime error of a problem that should be detectable at configure time.

Can you please eliminate this thrown error message, which will spam every time something is sent over kafka (correct?)

Actual behavior should be that when you specify a bad config `kafka.streams.log.compaction.strategy=(╯°□°）╯︵ ┻━┻` that the plugin at start time should say "Invalid log compaction strategy setting, switching to default value XYZ" and then at runtime use the correct default.

Reply:
@vga91 please add a check into the `validate` method of `KafkaConfiguration`